### DISCUSSION TITLE
Possibly a feature request - "Color Pie"

### DISCUSSION THREAD

**Discussion by richardsdev** [wojtryb/Shortcut-Composer#65] [Ideas]
I was thinking about ways to make selecting colours as slick as everything else is now that I have this addon installed.

Currently, I have the wide gamut colour selector mapped to my tab key. So I click tab, and get the quick selector thing. The behaviours of this little pop-up is slightly jarring compared to the nice interaction of your "temporary keys" feature. I would like to push and hold the tab key to have this pop-up shown, pick my color, then have it disappear when I release the key. I was thinking about that and then thought how nice it would be to have a "color pie menu", where you have a color selector in the middle, adjustment bars beneath it, and a configurable number of color slots around the circumference, which can be mapped to a palette. Maybe it could be a dynamic palette as well, where you can drop the active color into a slot on this pie to add one. The palette could be saved locally as well, so each time you open a krita file, you have the colours you were using ready to go.

**Comment by wojtryb** [comment]
Hi, and thanks for your feedback again.

While it's possible to set colors with the plugin, I've abandoned the idea of doing so in this plugin.
I know that existing solutions do not allow to set colors with "temporary key" logic, but the gain here is too small considering the expenses. Elements of krita GUI cannot be safely reused, so all types of color wheels, color selectors and so on need to be written for scratch to be used in a plugin.

Personally I use "Show color selector" pop-up, and I'm not sure if my own selector that could easily take months would be more convenient for me to use.

**Reply by richardsdev** [reply]
I see, well that is all reasonable. The idea wouldn't be a major improvement over what is already there, so aye, not worth it if it would take you a long time to develop. It would be very slick, but we can live without that :)

Thanks again for this great plugin

**Reply by fanialivio** [reply]
Hi there,

I understand that having a pop-up color selector does not worth the effort, and that the existing "Show color selector" already works fine,  but I have a question: is there a way to have "fixed" colors in a pie menu?
I use a very limited palette for my work, and I always have the "palette" docker visible. Do you think is possible to assign a created palette with a set of colors to a slot so that we can have a palette pie menu?

![Screenshot from 2023-10-23 10-40-35](https://github.com/wojtryb/Shortcut-Composer/assets/30428953/d856d08e-dd08-4ce5-9431-039a08b5cfea)

@wojtryb

**Reply by wojtryb** [reply]
Hi,

Sorry I haven't replied initially, but I've been quite busy lately, and wasn't working on the plugin that much.

This request is probably doable - though I would have to check if color palettes are available in python plugins.

If they are, that would be a rather big feature, as to have it implemented properly, it would have to work similar to brush presets with a "manual" and "tag (here palette)" modes.

That would be a cool feature, I'll remember about it, but I can't promise I'll have time for it.

**Reply by wojtryb** [reply]
Just letting you know that v1.6.0 introduces PieMenu for picking colors.

This is not exactly the feature you requested as there is no integration with krita palettes, but it does allow to store and pick colors using a pie.

**Reply by richardsdev** [reply]
Congrats on the new version. This new colour selector pie like it will be very useful!
